### ISSUE TITLE
PEAR.inc appears to be a PHP4 remnant

### ISSUE THREAD

**Issue by MichaelDeciso** [opnsense/core#2905]
PEAR.inc is full of PHP4 references, like:
```PHP
/**
 * Standard PEAR error class for PHP 4
 *
 * This class is supserseded by {@link PEAR_Exception} in PHP 5
 *
<snip>
 * @since      Class available since PHP 4.0.2
 */
class PEAR_Error
```
Looking further, it seems that (in `core` at least) only `IPv6.inc` actually still uses `PEAR.inc`, in order to do stuff like this:
```
PEAR::raiseError
```PHP
return PEAR::raiseError(NET_IPV6_NO_NETMASK_MSG,
                                        NET_IPV6_NO_NETMASK);
```
It seems to me that this could probably be replaced with throwing an exception. It seems to me that the only reason it doesn't is because PHP4 did not support exceptions.

So, in short, I would propose replacing PEAR::raiseError with throwing Exceptions, then getting rid of `PEAR.inc`.

**Comment by AdSchellevis** [comment]
Eventually, we should replace the IPv6.inc, it's old indeed.

**Comment by fichtner** [comment]
I'd say remove the "raiseError" and return null and remove PEAR.inc for 19.1 as the errors aren't raised because if they were we'd have application errors because they aren't cached? Long term there's net/pear-Net_IPv6 port which pulls in its own pear files via devel/pear.

**Comment by AdSchellevis** [comment]
@MichaelDeciso it looks like you're right here, let's kill PEAR.inc and return null. Eventually we should look for a replacement, but I'm not 100% sure what it should look like (Net_IPv6 doesn't look very active).

**Comment by fichtner** [comment]
super, thanks 👍 

There's a newer version in FreeBSD ports maintained by pfSense, but looking at the issues we've had with IPv6 related address handling and manipulation it's probably not worth the effort also because it's impossible to test the impact and spot possible regressions.

**Comment by fichtner** [comment]
(zero issues with faulty IPv6 address manipulation just to fill that implication)
